### ISSUE TITLE
jazz-midi critical dependancy

### ISSUE THREAD

**Issue by glend1** [djipco/webmidi#210]
everytime I load a page I get the following warning in the browser(chrome) `./node_modules/jazz-midi/index.js
Critical dependency: the request of a dependency is an expression`.

I am running on a Next.js development server. I don't know why this is happening. Doesn't seem to occur while running a production server.

**Comment by djipco** [comment]
Yes, I am aware of the warning but, so far, I haven't bee able to work around it. I'm guessing Next.js is using webpack, am I right?

**Comment by glend1** [comment]
yes

**Comment by djipco** [comment]
I made a little progress on that front. Here are the details (this is as much a reference for me as it is a follow-up to the issue):

* During compilation, Webpack (which is used by Next.js, React, etc.) tries to statically import all found dependencies. 
* When Webpack scans the WebMidi.js code, it finds a reference to the `jzz` module. This module is only needed when the library runs in Node.js but Webpack doesn't know that so it tries to compile it in the final bundle.
* The `jzz` module itself requires the `jazz-midi` module.
* This error is triggered because the `jazz-midi` module uses an expression to import the correct library according to the platform its running on (Mac, Win32 or Linux). This is fine in Node.js but Webpack does not like it because it prevents it from knowing what to import at compile time.

So, I guess, the solution would be either:

1. to tell Webpack to not include the `jzz` module when the final bundle is meant to be used in the browser or
2. to have a version of WEBMIDI.js for Node.js and another for the browser (which I would prefer to avoid)

I guess I will have to think about this...

**Comment by glend1** [comment]
If i don't use `jzz` then there isn't any way (that i've been able to find) to write tests for my application.

**Comment by djipco** [comment]
For testing purposes, I simply use  the CommonJS version (`webmidi.cjs.js`). I also use the `midi` package to create virtual MIDI ports (if you need that). Check out the [`test`](`webmidi.cjs.js`) folder for examples.

**Comment by djipco** [comment]
This is fixed in v3.0.8.
